### ISSUE TITLE
"Node must be provided when reporting error if location is not provided" error in 7.12.3

### ISSUE THREAD

**Issue by the-spyke** [yannickcr/eslint-plugin-react#2127]
Linting this code with `7.12.3` gives me an error `"AssertionError [ERR_ASSERTION]: Node must be provided when reporting error if location is not provided"`:

```js
import React from "react";

export default class Controller extends React.Component {
    handleAdd = id => {
        this.setState((state, { name }) => {
            const item = this.buildItem(id);
        });
    };
}
```

Looks similar to #2095. Works fine in `7.11.0`.

**Comment by the-spyke** [comment]
Thanks, ESLint works fine now in my monorepo.

**Comment by alexzherdev** [comment]
@ljharb looking at your fix, it seems that we're still incorrectly detecting something as a prop being used (where in reality this example is not using any props). I'll see if I get time to figure it out.

**Comment by ljharb** [comment]
Probably so, but this fix passed the test :-) improvements welcome!
